src/libraries/System.Threading.Tasks.Dataflow/src/Blocks/TransformManyBlock.IAsyncEnumerable.cs

```diff
@@ -0,0 +1,247 @@
+                    _target.Complete(exc, dropPendingMessages: true, storeExceptionEvenIfAlreadyCompleting: true, unwrapInnerExceptions: false);
+                }
+            }
+            finally
+            {
+                // Let the target know that one of the asynchronous operations it launched has completed.
+                _target.SignalOneAsyncMessageCompleted();
+            }
+        }
+
+        /// <summary>
+        /// Stores the output items, either into the reordering buffer or into the source half.
+        /// Ensures that the bounding count is correctly updated.
+        /// </summary>
+        /// <param name="messageWithId">The message with id.</param>
+        /// <param name="outputItems">The output items to be persisted.</param>
+        private async Task StoreOutputItemsAsync(
+            KeyValuePair<TInput, long> messageWithId, IAsyncEnumerable<TOutput>? outputItems)
+        {
+            // If there's a reordering buffer, pass the data along to it.
+            // The reordering buffer will handle all details, including bounding.
+            if (_reorderingBuffer != null)
+            {
+                await StoreOutputItemsReorderedAsync(messageWithId.Value, outputItems).ConfigureAwait(false);
+            }
+            // Otherwise, output the data directly.
+            else if (outputItems != null)
+            {
+                await StoreOutputItemsNonReorderedWithIterationAsync(outputItems).ConfigureAwait(false);
+            }
+            else if (_target.IsBounded)
+            {
+                // outputItems is null and there's no reordering buffer
+                // and we're bounding, so decrement the bounding count to
+                // signify that the input element we already accounted for
+                // produced no output
+                _target.ChangeBoundingCount(count: -1);
+            }
+            // else there's no reordering buffer, there are no output items, and we're not bounded,
+            // so there's nothing more to be done.
+        }
+
+        /// <summary>Stores the next item using the reordering buffer.</summary>
+        /// <param name="id">The ID of the item.</param>
+        /// <param name="item">The async enumerable.</param>
+        private async Task StoreOutputItemsReorderedAsync(long id, IAsyncEnumerable<TOutput>? item)
+        {
+            Debug.Assert(_reorderingBuffer != null, "Expected a reordering buffer");
+            Debug.Assert(id != Common.INVALID_REORDERING_ID, "This ID should never have been handed out.");
+
+            // Grab info about the transform
+            TargetCore<TInput> target = _target;
+            bool isBounded = target.IsBounded;
+
+            // Handle invalid items (null enumerables) by delegating to the base
+            if (item == null)
+            {
+                _reorderingBuffer.AddItem(id, null, false);
+                if (isBounded) target.ChangeBoundingCount(count: -1);
```

Comment:
Nit: Add Braces

*Personally, I also prefer the style in such cases. But the runtime coding style specifies that you should use braces.*
https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md (18.)